bobcat/bb-cumber/src/main/java/com/cognifide/qa/bb/cumber/Bobcumber.java

```diff
@@ -161,9 +130,31 @@ private void closeWebDriverPool() {
     }
   }
 
-  private boolean isFeatureFileEmpty() throws IOException {
-    try (BufferedReader br = new BufferedReader(new FileReader(file))) {
-      return br.readLine() == null;
+  private void shutdownRerun(RunNotifier notifier) {
+    double percentageOfFailedTests = statisticsHelper.getPercentageOfFailedTests(statisticsFile);
+    int failedTestsNumber = statisticsHelper.getNumberOfFailedTests(statisticsFile);
+    if (failedTestsNumber == 0) {
+      notifier.fireTestFinished(Description.EMPTY);
+    } else if (percentageOfFailedTests > maxFailedTestPercentage) {
+      String failureMessage = "Percentage of failed tests was bigger than " + maxFailedTestPercentage + ".";
+      Failure failure = new Failure(Description.createSuiteDescription(failureMessage), new TooManyTestsToRerunException(failureMessage));
+      notifier.fireTestFailure(failure);
     }
   }
+
+  private boolean canRerunFailedTests() {
+    int failedTestsNumber = statisticsHelper.getNumberOfFailedTests(statisticsFile);
+    double percentageOfFailedTests = statisticsHelper.getPercentageOfFailedTests(statisticsFile);
+    boolean zeroTests = (failedTestsNumber == 0);
+    boolean haveTooManyTests = (percentageOfFailedTests > maxFailedTestPercentage);
+    return !(zeroTests || haveTooManyTests);
```

Comment:
`!zeroTests && !haveTooManyTests` would be clearer I think? Perhaps renaming `zeroTests` would be ok too